### ISSUE TITLE
Java version compatibility between Jasper Reports and Dynamic Reports

### ISSUE THREAD

**Issue by celyo** [dynamicreports/dynamicreports#181]
Hello,

Dynamic Reports is following Jasper Reports semantic version numbering for many years.  This practically means that Dynamic Reports 6.12.x is using Jasper Reports 6.12.x, Dynamic Reports 6.20.x is using Jasper Reports 6.20.x, etc.

Unfortunately it's not the same when it comes to Java versions:

- Jasper Reports 6.12.x is compatible with Java 8 and can be used in Java 11 -> Dynamic Reports 6.12.x supports the same versions
- Jasper Reports 6.20.x is compatible with Java 8 and can be used in Java 11 -> Dynamic Reports 6.20.x requires Java 17+

Keeping the compatibility is important for many existing projects (still compiling with Java 11), using both Dynamic Reports & Jasper Reports. Especially considering that Jasper Reports 7 was release 1-2 months ago and it's not so widely used yet. One more argument for keeping java version compatibility could also be security vulnerabilities in 3th party dependencies in Jasper Reports, which requires upgrade to more recent version of Jasper Reports 6 and Dynamic Reports 6.

**Comment by jan-moxter** [comment]
Hello and thanks for the input. I am surprised about the input, because I literally started to think about this project as my small little project that I keep alive only due to using dynamic reports now for years in the main project (the one that pays the bills :-) ) . It was abandoned from the original developer  ( we assume that he is no longer with us) and via a backup copy resurrected and currently I am the only one left working  on it occasionally.
So in short terms, if my main project switches java version I just follow along without giving it to much thought.....  and I literally thought that no-one else is using it   .....

Having different versions using different java requirements.... seems like a lot of efforts, but if you have an idea/concept/strategy so that we can do that. Well I am open to hear it :+1:

**Comment by celyo** [comment]
First thanks a lot for your effort supporting this project! Having in mind that Jasper Reports 7 is already requiring Java 17, all projects which want to use latest Jasper Reports have to switch to 17 as well relatively soon.

Meanwhile what comes to my mind ... even using Java 17 in your machine there is possibility to compile/release the library for 8 or 11 if source code allows it. Currently it's possible, I've compiled the project with Java 11 by just changing source & target versions in maven-compiler-plugin. This will allows all of us to continue using Dynamic Reports in our projects and being relatively up to date with both Dynamic & Jasper Reports while working on migration to Java 17.

**Comment by ccorneli** [comment]
Hi @jan-moxter,  I'd also like to request java 1.8 compatibility for DynamicReports.
As celyo mentioned, I expect you would just need to change the source/target in the maven pom.xml file.  I'd also suggest that the build configuration and platform requirements try to mirror JasperReports as closely as possible for compatibility reasons, and it looks like the JasperReport project is still configured to support java 8 based on thier [pom-parent.xml#L96](https://github.com/TIBCOSoftware/jasperreports/blob/d736357b5d6d08294e7604ad511f68a9798689bf/pom-parent.xml#L96)


@celyo, the only thing I could find about Jasper Reports 'already requiring Java 17' was related to using plug-ins with the studio.  Unfortunately I can't find the reference at the moment.. but for the base library, I didn't find anything mentioned in the v7 release documentation or upgrade guide.  Infact it seems to still be a priority to keep 1.8 compatibility as mentioned in the [Readme.md](https://github.com/TIBCOSoftware/jasperreports/blob/d736357b5d6d08294e7604ad511f68a9798689bf/README.md?plain=1#L99)

>From time to time, verifying that the core and extensions artifacts are still compatible with JDK version 1.8 is needed and this is done by turning on the enforcer plugin while building these artifacts

Brief Background:  I stumbled here while trying to address a vulnerabilty in our project with jFreechart 1.x, starting by trying to find a way to upgrade the nested dependency to 1.5.5, then to upgrade JasperReports... We currently use DynamicReports v6.12.1 but ran into the "class file major version 61" from this project after trying to upgrade to 6.20.x 

Related: JasperReport [Issue #373](https://github.com/TIBCOSoftware/jasperreports/issues/373)

**Comment by jvosloo** [comment]
> Hello and thanks for the input. I am surprised about the input, because I literally started to think about this project as my small little project that I keep alive only due to using dynamic reports now for years in the main project (the one that pays the bills :-) ) . It was abandoned from the original developer ( we assume that he is no longer with us) and via a backup copy resurrected and currently I am the only one left working on it occasionally. So in short terms, if my main project switches java version I just follow along without giving it to much thought..... and I literally thought that no-one else is using it .....
> 
> Having different versions using different java requirements.... seems like a lot of efforts, but if you have an idea/concept/strategy so that we can do that. Well I am open to hear it 👍

We’re also still using this library…. With Java 17 for a while now already. Thanks for keeping on maintaining it 🙏

**Comment by jan-moxter** [comment]
Looking at the current jasper report version, that I am working on in the 7.0.0. development branch and it seems they still decided to compile it with java 8.  :thinking: 
I am not planning to do anything else than migrate so I could maintain the java 8 stile .... So it would be easier backward compatible and it would also work in all following ones.
